cmd/serve.go

```diff
@@ -110,43 +122,40 @@ func doDispatch(ctx context.Context, addr string, remotes []*node) error {
 		return err
 	}
 
-	//fmt.Println(string(respBody))
-
 	return json.Unmarshal(respBody, &remotes)
 }
 
-func listenDispatch(ctx context.Context) func(w http.ResponseWriter, r *http.Request) {
-	return func(w http.ResponseWriter, r *http.Request) {
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte(err.Error()))
-			return
-		}
+func listenDispatch(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write(errorify(err))
+		return
+	}
 
-		p := &[]*node{}
-		if err := json.Unmarshal(body, p); err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte(err.Error()))
-			return
-		}
+	p := &[]*node{}
+	if err := json.Unmarshal(body, p); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write(errorify(err))
+		return
+	}
 
-		for _, px := range *p {
-			if err := doPerf(ctx, px); err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				w.Write([]byte(err.Error()))
-				return
-			}
-		}
-		respBody, err := json.Marshal(p)
-		if err != nil {
+	for _, px := range *p {
+		if err := doPerf(ctx, px); err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte(err.Error()))
+			w.Write(errorify(err))
 			return
 		}
-		w.WriteHeader(http.StatusOK)
-		w.Write(respBody)
 	}
+	respBody, err := json.Marshal(p)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write(errorify(err))
+		return
+	}
+	w.WriteHeader(http.StatusOK)
```

Comment:
This is implicit not necessarily needed since w.Write() adds StatusOK automatically.

Reply:
You only need it to designate errors.